src/contribute/contribution-guide.md

```diff
@@ -58,30 +58,30 @@ If interested, you can also join the other [mailing lists]({{ site.baseurl
 We use the Apache Software Foundation's
 [JIRA](https://issues.apache.org/jira/browse/BEAM) as an issue tracking and
 project management tool, as well as a way to communicate among a very diverse
-and distributed set of contributors. To be able to gather feedback, avoid
-frustration, and avoid duplicated efforts all Beam-related work should be
-tracked there.
+and distributed set of contributors. To avoid duplicated efforts,
+significant Beam-related work should be tracked there.
 
 If you do not already have an Apache JIRA account, sign up
 [here](https://issues.apache.org/jira/).
 
 If a quick
 [search](https://issues.apache.org/jira/issues/?jql=project%3DBEAM%20AND%20text%20~%20%22the%20thing%20I%20want%20to%20contribute%22)
 doesn’t turn up an existing JIRA issue for the work you want to contribute,
-create it. Please discuss your idea with a committer or the [component
+feel free to create one. Please discuss your idea with a committer or the [component
 lead](https://issues.apache.org/jira/browse/BEAM/?selectedTab=com.atlassian.jira.jira-projects-plugin:components-panel)
 in JIRA or, alternatively, on the developer mailing list.
 
 If there’s an existing JIRA issue for your intended contribution, please
-comment about your intended work. Once the work is understood, a committer will
+comment about your intended work. A committer can
 assign the issue to you. (If you don’t have a JIRA role yet, you’ll be added to
-the “contributor” role.) If an issue is currently assigned, please check with
-the current assignee before reassigning.
+the “contributor” role.) If an issue is currently assigned, it is generally
+good practice to at least reach out the current assignee before reassigning.
 
-For moderate or large contributions, you should not start coding or writing a
-design document unless there is a corresponding JIRA issue assigned to you for
-that work. Simple changes, like fixing typos, do not require an associated
-issue.
+For moderate or large contributions, we recommend engaging on the developer
```

Comment:
Here, I would even say "hey the dev list is great and you should talk to us about what you are interested in" or drop the whole paragraph just to cut size.